docs/api/connection.rst

```diff
@@ -9,19 +9,21 @@ API
 Connection
 ==========
 
-.. js:function:: connect(options)
+.. js:function:: connect(dsn, options)
```

Comment:
Paul, I don't like breaking the API just yet. I'd do this differently: let's make `options` to be of type `ConnectConfig | string | null`.

We can then gradually introduce "instances" - they'll be a separate JS type that we will be able to check with an `instanceof` - and ban plain objects (currently `ConnectConfig`)

Reply:
Well, then you can't set timeout. Or the API would look like:
```
connect(name_or_dsn_or_options: string | ConnectOptions | null, options: ConnectOptions | null)
```
(and then throw error if both are objects?) I.e. jQuery-age libraries had a lot of API looking like this, but in 2020... 

And then what to do with `createPool`? This function without options is next to useless.

Reply:
As discussed on the call, let's try:

Support all `connect(instanceName)`, `connect(ConnectConfig)` and `connect(instanceName, ConnectConfig)` use cases. The signature will likely look like this:

`connect(ConnectConfig | string | null, ConnectConfig | null)`

The code will need to validate that only one `ConnectConfig` is actually passed.

We'll start issuing deprecation warnings once we fully agree on the new instance API and start implementing it.